### ISSUE TITLE
GPU inference ignores a model's separate output head (uses the tied embedding for all models)

### ISSUE THREAD

**Issue by MBrekhof** [Integral2u/SharpMind#38]
`GpuInferenceEngine.LmHead` always projects the final-normed hidden state against the token embedding (`_embedding` / `_embeddingRaw`). The CPU `Transformer` uses the separate `LmHead` tensor when the model has one and only falls back to the tied embedding when it does not. `GpuInferenceEngine.ValidateSupported` accepts models with a separate output head, so an untied model runs on the GPU path and produces logits from the wrong matrix, silently.

Affected: any GGUF with `output.weight` distinct from `token_embd.weight` (the larger checkpoints; the small Qwen2 ones are tied, which is why the existing parity tests pass).

**Repro** (synthetic, no checkpoint needed):

```csharp
var cfg = new ModelConfig { VocabSize = 256, HiddenDim = 64, NumLayers = 2, NumHeads = 4, NumKvHeads = 2, FfnDim = 128, MaxSeqLen = 64, Architecture = "qwen2" };
var config = SharpMindConfig.Llama with { Hardware = HardwareTier.Scalar };
var weights = ModelFactory.CreateForTraining(cfg, config);
WeightInitializer.InitializeRandomly(weights, 9001);
weights.SetLmHead(new Tensor<float>(cfg.VocabSize, cfg.HiddenDim));   // all-zero separate head

using var model = ModelFactory.CreateTrainingTransformer(weights, config);
using var device = GpuDevice.Create();
using var engine = new GpuInferenceEngine(device, model, config, 64, 32);
int[] ids = [65, 66, 67, 32, 68, 69];

// CPU: Prefill.ForwardLastLogitsChunked(model, caches, ids, workspace) -> every logit is exactly 0
// GPU: engine.Prefill(ids)                                              -> max |logit| = 0.827 (same as the tied model)
```

With the same weights and no `SetLmHead` call, CPU and GPU agree to 1.8e-7, so the rest of the forward pass is fine; it is only the head.

**Suggested fix:** upload the separate head when present (F32 via `Gemm`, Q8_0 via `DequantMatmul`, mirroring the embedding branches) and use it in `LmHead`; or, until then, have `ValidateSupported` reject models whose `LmHead` is non-null so they fall back to CPU instead of returning wrong logits.

Observed on OpenCL (gfx1152); the code path is backend-independent.

**Comment by Integral2u** [comment]
Thank you.
Update in commit "fix: GPU inference uses the separate LM head, not the tied embedding" 09e8c9f1
